### ISSUE TITLE
trained for 3 days but results are very strange

### ISSUE THREAD

**Issue by hassanrazakhalid** [lizhengwei1992/Semantic_Human_Matting#44]
hi
i trained on company dataset for almost 3-4 days on GTX 1080TI.  Please see the log file. Error was keep getting low as expected but when i tested even on trained images. It is behaving strangely
[log.txt](https://github.com/lizhengwei1992/Semantic_Human_Matting/files/4490341/log.txt)

Please see the orignal image and result after training. Expect for a few white dashes nothing is there.
![out](https://user-images.githubusercontent.com/15266641/79519479-702cdb80-806d-11ea-8683-7aeca2961bf9.png)
![1803151818-00000004](https://user-images.githubusercontent.com/15266641/79519497-78851680-806d-11ea-961d-e631d5d7dd13.jpg)

Any idea what i did wrong here. Note i generated alphas by seprating  the alpha channel in provided masks and not KNN. Sample orignal GT is
![1803151818-00000008](https://user-images.githubusercontent.com/15266641/79519614-cc8ffb00-806d-11ea-998d-da551cd09e24.png)

**Comment by Plwy** [comment]
if you don't have ground truth alpha matte, you need to generate. The mask is just a coarse seg of the person ,can not represent the ground truth alpha matte .you need to use the mask to generate the trimap ,then use trimap and origin image as the input of knn-matting to generate alpha matte.We use the alpha matte that knn-matting generate as the approximation of ground truth alpha matte.

**Comment by ghazni123** [comment]
Hi 

I am getting the same results (i.e. trained model is not doing matting at all). My process is as:

1: original image
2: manually calculated alpha-matte in photoshop
3: trimap generated from gen_trimap code in repo (by extracting alpha from foreground image as mask to input gen_trimap)

I have used all the default values in scripts. My all images (input, alpha-matte and trimap) are approximately in 1000x700 resolution. Since patch-size is 320 so could this be the cause for not learning? that patch size is so small and might not be picking right portions in the training dataset.

Other than that what should be the parameters (like nEpochs) for SOTA results and what is expected loss after training?    

Thank you.

Log is attached and sounds like it is not learning.
[log.txt](https://github.com/lizhengwei1992/Semantic_Human_Matting/files/5582234/log.txt)
